WordPress/openverse-frontend
# Update Inter font and Playwright version

## Fixes


Fixes #1781 by @obulat
Fixes #1754 by @obulat

## Description


These two changes seem to be unrelated at first, but updating Playwright version required updating the fonts to fix the snapshots.

This PR adds a variable Inter font for platforms that support variable fonts, and more modern `woff` and `woff2` font files instead of the `ttf` font files. It also updates the Playwright version to the latest 1.28.1, and uses a `jammy` Docker image (replacing the older `focal` image).


@panchovm, could you check if the fonts look right by running the app locally? The snapshots' fonts look different than what I see on my Mac, or on a Linux machine. Here, the first snapshot is from the current implementation, and the second one is what I see with this PR's changes. The letters look less squished together and more rounded.

<img width="322" alt="Screenshot 2022-12-03 at 4 44 42 PM" src="https://user-images.githubusercontent.com/15233243/205446681-555a269f-eeab-471d-accf-3afd134cf41e.png">

<img width="325" alt="Screenshot 2022-12-03 at 4 45 20 PM" src="https://user-images.githubusercontent.com/15233243/205446674-b9e586ed-db6f-4c8e-878b-c41980c0234d.png">
Several interesting things about this PR:


- We did not have a `400` font, instead we only had the `500` weighted one. I don't know how it worked.
- We probably had an older version of the Inter font because it looked differently. Now, the fonts that are used seem to look more like the Figma mockups.
- The font rendering on different systems is complex, and fonts look differently on different platforms. So, the snapshots we get in a Linux Docker image are different from what I see on a Mac.
-  Inter, the font we are using, does not have Arabic support. So, in the Arabic version, we are using a fallback sans-serif font.

## Testing Instructions

Run the app and check that fonts match the Figma snapshots.
The CI should also pass, and specifically RTL snapshots should not have any `NULL` characters or breakages.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
